app/models/spree/app_configuration.rb

```diff
@@ -105,7 +105,6 @@ class AppConfiguration < Preferences::Configuration
     preference :embedded_shopfronts_whitelist, :text, default: nil
 
     # Legal Preferences
-    preference :footer_tos_url, :string, default: "/Terms-of-service.pdf"
```

Comment:
Oh, this is going a step further than I anticipated. I was talking about the default being obsolete. But some instances may still use this old setting with valid URLs. I forgot that option. The spree_preferences table does have an updated_at attribute which we could use. But, hey, let's check first if we can get rid of the altogether as you are suggesting now.

```
$ sql='SELECT * FROM spree_preferences WHERE key = $$/spree/app_configuration/footer_tos_url$$ LIMIT 1;'
$ ansible all-prod -u openfoodnetwork -a "psql -h localhost openfoodnetwork ofn_user -c '$sql'"

openfoodnetwork.org.au | CHANGED | rc=0 >>
  id   |         value         |        created_at         |         updated_at         |                   key                   | value_type 
-------+-----------------------+---------------------------+----------------------------+-----------------------------------------+------------
 13790 | /Terms-of-service.pdf | 2019-01-25 04:55:28.99796 | 2020-02-24 22:31:24.286698 | /spree/app_configuration/footer_tos_url | string
(1 row)
openfoodnetwork.org.nz | CHANGED | rc=0 >>
 id |                    value                     |         created_at         |         updated_at         |                   key                   | value_type 
----+----------------------------------------------+----------------------------+----------------------------+-----------------------------------------+------------
 31 | https://openfoodnetwork.nz/terms-of-service/ | 2019-09-24 07:41:12.235821 | 2019-12-06 09:57:28.479701 | /spree/app_configuration/footer_tos_url | string
(1 row)
openfoodnetwork.ie | CHANGED | rc=0 >>
 id |                   value                    |         created_at         |         updated_at         |                   key                   | value_type 
----+--------------------------------------------+----------------------------+----------------------------+-----------------------------------------+------------
 30 | --- https://about.openfoodnetwork.ie/terms+| 2020-03-28 18:40:15.753541 | 2020-08-21 20:56:33.074604 | /spree/app_configuration/footer_tos_url | string
    | ...                                       +|                            |                            |                                         | 
    |                                            |                            |                            |                                         | 
(1 row)
openfoodnetwork.org.uk | CHANGED | rc=0 >>
  id  |                                       value                                        |         created_at         |         updated_at         |                   key                   | value_type 
------+------------------------------------------------------------------------------------+----------------------------+----------------------------+-----------------------------------------+------------
 8538 | https://drive.google.com/file/d/1ne3GqiXnzTw6tWD484agzWqhlerJzH8p/view?usp=sharing | 2018-09-08 10:48:37.934328 | 2020-06-04 07:57:29.594535 | /spree/app_configuration/footer_tos_url | string
(1 row)
openfoodnetwork.ca | CHANGED | rc=0 >>
  id  |            value            |         created_at         |         updated_at         |                   key                   | value_type 
------+-----------------------------+----------------------------+----------------------------+-----------------------------------------+------------
 2023 | --- "/Terms-of-service.pdf"+| 2018-10-08 03:20:50.636275 | 2020-11-03 21:29:56.958475 | /spree/app_configuration/footer_tos_url | string
      |                             |                            |                            |                                         | 
(1 row)
app.katuma.org | CHANGED | rc=0 >>
 id  |            value            |         created_at         |         updated_at         |                   key                   | value_type 
-----+-----------------------------+----------------------------+----------------------------+-----------------------------------------+------------
 926 | --- "/Terms-of-service.pdf"+| 2018-12-03 13:16:17.696466 | 2020-11-06 18:08:07.512145 | /spree/app_configuration/footer_tos_url | string
     |                             |                            |                            |                                         | 
(1 row)
csicsoka.org | CHANGED | rc=0 >>
 id | value | created_at | updated_at | key | value_type 
----+-------+------------+------------+-----+------------
(0 rows)
openfoodnetwork.de | CHANGED | rc=0 >>
 id  |            value            |         created_at         |         updated_at         |                   key                   | value_type 
-----+-----------------------------+----------------------------+----------------------------+-----------------------------------------+------------
 300 | --- "/Terms-of-service.pdf"+| 2019-04-16 10:19:36.117812 | 2020-12-04 18:43:16.129764 | /spree/app_configuration/footer_tos_url | string
     |                             |                            |                            |                                         | 
(1 row)
openfoodnetwork.net | CHANGED | rc=0 >>
 id  |            value            |         created_at         |        updated_at         |                   key                   | value_type 
-----+-----------------------------+----------------------------+---------------------------+-----------------------------------------+------------
 527 | --- "/Terms-of-service.pdf"+| 2018-11-16 03:58:49.307589 | 2020-11-16 13:25:56.34786 | /spree/app_configuration/footer_tos_url | string
     |                             |                            |                           |                                         | 
(1 row)
openfoodnetwork.be | CHANGED | rc=0 >>
 id |         value         |         created_at         |         updated_at         |                   key                   | value_type 
----+-----------------------+----------------------------+----------------------------+-----------------------------------------+------------
 41 | /Terms-of-service.pdf | 2018-10-29 20:04:44.172964 | 2019-10-28 13:59:52.575425 | /spree/app_configuration/footer_tos_url | string
(1 row)
coopcircuits.fr | CHANGED | rc=0 >>
  id  |                           value                           |         created_at         |        updated_at         |                   key                   | value_type 
------+-----------------------------------------------------------+----------------------------+---------------------------+-----------------------------------------+------------
 3992 | --- https://apropos.coopcircuits.fr/informations-legales/+| 2018-10-09 13:03:37.022249 | 2020-11-16 09:32:46.59564 | /spree/app_configuration/footer_tos_url | string
      | ...                                                      +|                            |                           |                                         | 
      |                                                           |                            |                           |                                         | 
(1 row)
```


```
$ sql='SELECT COUNT(*) FROM terms_of_service_files'
$ ansible all-prod -u openfoodnetwork -a "psql -h localhost openfoodnetwork ofn_user -c '$sql'"

openfoodnetwork.org.au | CHANGED | rc=0 >>
 count 
-------
     2
(1 row)
openfoodnetwork.org.nz | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.ca | CHANGED | rc=0 >>
 count 
-------
     1
(1 row)
openfoodnetwork.org.uk | CHANGED | rc=0 >>
 count 
-------
     1
(1 row)
app.katuma.org | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
csicsoka.org | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.de | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
openfoodnetwork.net | CHANGED | rc=0 >>
 count 
-------
     1
(1 row)
coopcircuits.fr | CHANGED | rc=0 >>
 count 
-------
     1
(1 row)
openfoodnetwork.be | CHANGED | rc=0 >>
 count 
-------
     1
(1 row)
openfoodnetwork.ie | CHANGED | rc=0 >>
 count 
-------
     0
(1 row)
```

Instances without uploaded file:
- openfoodnetwork.org.nz -> https://openfoodnetwork.nz/terms-of-service/
- app.katuma.org -> /Terms-of-service.pdf
- csicsoka.org -> nil, default
- openfoodnetwork.de -> /Terms-of-service.pdf
- openfoodnetwork.ie -> https://about.openfoodnetwork.ie/terms

So we have two instances which would need to upload their terms of service as PDF.

Maybe a question for @openfoodfoundation/train-drivers-product-owners: Do we want to continue to support a URL as platform terms of service (2 instances) or do we want all instances to use an uploaded PDF file (6 instances)?

Only PDF is less code to maintain but possibly not as good UX.